lib/snmplib/snmp_vars.c

```diff
@@ -512,7 +512,7 @@ snmp_var_DecodeVarBind(u_char * Buffer, int *BufLen,
             bufp = asn_parse_string(DataPtr, &ThisVarLen,
                                     &Var->type, Var->val.string,
                                     &Var->val_len);
-            if (Var->val_len < terminatorPos) {
+            if (Var->val_len < terminatorPos || terminatorPos < 0) {
                 terminatorPos = Var->val_len;
             }
             Var->val.string[terminatorPos] = '\0';
```

Comment:
> zero-length OCTET STRING should not access a -1 buffer

I agree. However, AFAICT, terminatorPos becomes negative when val_len is zero. If val_len is zero, then zero val.string bytes may be allocated by xmalloc() above[^1], terminatorPos becomes zero (in PR code) and, technically, this code line accesses a val.string character that may not be allocated...

Please refactor to avoid xmalloc(0) and always allocate/terminate val.string, even if TheVarLen is zero.

[^1]: [AFAICT](https://stackoverflow.com/a/1073167), malloc(0) return value is implementation defined: It may be either nil or a non-nil pointer to a zero-size area (i.e. one must not dereference that pointer).

Reply:
Should be fine now.